crates/storage/src/interface.rs

```diff
@@ -102,11 +145,23 @@ impl Interface {
     ///
     /// # Errors
     ///
-    /// If an error occurs when interacting with the storage system, an error
-    /// will be returned.
-    ///
-    pub fn save(&self, _id: Id, _element: &Element) -> Result<(), StorageError> {
-        unimplemented!()
+    /// If an error occurs when serialising data or interacting with the storage
+    /// system, an error will be returned.
+    ///
+    pub fn save(&self, id: Id, element: &Element) -> Result<(), StorageError> {
+        // TODO: It seems fairly bizarre/unexpected that the put() method is sync
+        // TODO: and not async. The reasons and intentions need checking here, in
+        // TODO: case this save() method should be async and wrap the blocking call
+        // TODO: with spawn_blocking(). However, this is not straightforward at
+        // TODO: present because Slice uses Rc internally for some reason.
+        self.store
+            .write()
+            .put(
```

Comment:
are we losing atomicity?

Reply:
How? 🤔

Reply:
So previously, if a transaction (execution request) fails, all of it's changes to storage are rolled back.

Just checking if this is behaviour you intend to retain under this model.

Reply:
Ah, I see now. So at present there is no context of that at all, and it's unclear if there will be. There is only one operation right now, hence it is inherently atomic. If/when that expands to have multiple operations (which may well happen), then yes such things would naturally need to be wrapped in a transaction. So I don't think this applies here.

Reply:
Approving to unblock, but can you clarify what you mean by:

> There is only one operation right now